lib/practice_exercises.rb

```diff
@@ -1,13 +1,45 @@
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n)
+# Space Complexity: O(n)
 def is_palindrome(string)
+  array = []
+  string = string.upcase
+  array = string.scan /[a-zA-Z]/
+  half = array.length / 2
+  back_half = []
+  half.times {back_half << array.pop}
+  if array.length != back_half.length
+    array.pop
+  end
+  if string.length == 1 || string == ""
+    return true
+  elsif array == back_half
+    return true
+  else
+    return false
+  end
+
   raise NotImplementedError, "Not implemented yet"
 end
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n^2)
+# Space Complexity: O(n)
 def longest_prefix(strings)
```

Comment:
I would say the time complexity is O(n * m) where m is the length of the shortest word and n being the number of strings.  Otherwise 👍